cpp/src/io/parquet/page_hdr.cu

```diff
@@ -405,7 +405,7 @@ __global__ void __launch_bounds__(128)
             case PageType::DATA_PAGE_V2:
               index_out = num_dict_pages + data_page_count;
               data_page_count++;
-              bs->page.flags = 0;
+              bs->page.flags = PAGEINFO_FLAGS_V2;
```

Comment:
They don't _have_ to be OR'd because the 3 states we have are orthogonal...the page is either a dictionary page (flags == 1), regular data page (flags == 0), or V2 data page (flags == 2). But given that they sort of comprise a mask, they probably should be OR'd just in case other flags are added in the future (can't really think of any imminent ones right now). We'd just have to move the initialization of `flags` outside the switch.

I can make that change if you'd like.